src/pudl/transform/ferc1.py

```diff
@@ -188,52 +189,60 @@ def get_ferc1_dbf_xbrl_glue(dbf_table_name: str) -> pd.DataFrame:
         row_map.sched_table_name == dbf_table_name,
         ["report_year", "row_number", "row_type", "xbrl_column_stem"],
     ]
-    # Indicate which rows have unmappable headers in them, to differentiate them from
-    # null values in the exhaustive index we create below.
-    # We need the ROW_HEADER sentinel value so we can distinguish
-    # between two different reasons that we might find NULL values in the
-    # xbrl_column_stem field:
-    # 1. It's NULL because it's between two valid mapped values (the NULL was created
-    #    in our filling of the time series) and should thus be filled in, or
-    # 2. It's NULL because it was a header row in the DBF data, which means it should
-    #    NOT be filled in. Without the HEADER_ROW value, when a row number from year X
-    #    becomes associated with a non-header row in year X+1
-    #    the ffill will keep right on filling, associating all of the new header rows
-    #    with the value of xbrl_column_stem that was associated with the old row number.
-    row_map.loc[
-        (row_map.row_type == "header") & (row_map.xbrl_column_stem.isna()),
+    return row_map
+
+
+def fill_dbf_to_xbrl_map(df: pd.DataFrame, dbf_years: Iterable[int]) -> pd.DataFrame:
```

Comment:
I broke the `read_` and `fill_` functions apart and made this one take `dbf_years` as an argument to make testing easier.